include/dca/phys/dca_step/cluster_solver/ctint/structs/ct_int_matrix_configuration.hpp

```diff
@@ -28,22 +28,44 @@ namespace ctint {
 
 // Expansion term.
 struct Vertex {
+  Vertex() = default;
+  Vertex(bool _aux_spin, ushort _interaction_id, std::uint64_t _tag, double _tau)
+      : aux_spin(_aux_spin), interaction_id(_interaction_id), tag(_tag), tau(_tau) {
+    spins.fill(-1);
+    matrix_config_indices.fill(-1);
+  }
+
   bool aux_spin;
   ushort interaction_id;
   std::uint64_t tag;
   double tau;
 
+  // Reference to the matrix config
+  std::array<std::uint8_t, 2> spins;
+  std::array<unsigned, 2> matrix_config_indices;
+
+  // Marks if this vertex is part of the accepted configuration.
+  bool annihilatable = false;
+
   bool operator==(const Vertex& b) const {
     return aux_spin == b.aux_spin && interaction_id == b.interaction_id && tau == b.tau;
   }
 };
 
+struct ConfigRef {
+  ConfigRef() = default;
+  ConfigRef(unsigned _config_id, std::uint8_t _leg_id) : config_id(_config_id), leg_id(_leg_id) {}
+
+  unsigned config_id;   // Index of the interaction in the SolverConfiguration.
+  std::uint8_t leg_id;  // Stores if this is the first or second leg of the interaction.
```

Comment:
So are the only valid values 1 and 2 or is there more to it?

Reply:
0 and 1. Is the comment confusing? I updated it in 727f88b2